### PR TITLE
Support for `textDocument/documentSymbol`

### LINKED ISSUE #22: Support for `textDocument/documentSymbol`
Return list of properties declared in application.properties as Symbol to fill the outline.

image

### PR DESCRIPTION
**Pull request by angelozerr**
Support for `textDocument/documentSymbol`

Fixes #22

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

File: quarkus.ls/com.redhat.quarkus.ls/src/main/java/com/redhat/quarkus/ls/QuarkusTextDocumentService.java [hunk 1]
```diff
@@ -143,6 +147,20 @@ public void didSave(DidSaveTextDocumentParams params) {
 		});
 	}
 
+	@Override
+	public CompletableFuture<List<Either<SymbolInformation, DocumentSymbol>>> documentSymbol(
+			DocumentSymbolParams params) {
+		return getPropertiesModel(params.getTextDocument(), (cancelChecker, document) -> {
+			return getQuarkusLanguageService().findSymbolInformations(document, cancelChecker) //
+					.stream() //
+					.map(s -> {
```
Comment (fbricon) [inline quarkus.ls/com.redhat.quarkus.ls/src/main/java/com/redhat/quarkus/ls/QuarkusTextDocumentService.java]: .map(Either::forLeft)

**Comment by fbricon** [comment]
I'd remove the empty lines from the outline. 
Don't need the values I think

File: quarkus.ls/com.redhat.quarkus.ls/src/main/java/com/redhat/quarkus/ls/QuarkusTextDocumentService.java (hunk 1 above)
Comment (angelozerr) [reply quarkus.ls/com.redhat.quarkus.ls/src/main/java/com/redhat/quarkus/ls/QuarkusTextDocumentService.java]: I tried it to do that, but I had compilation problem?

**Comment by angelozerr** [comment]
> I'd remove the empty lines from the outline. Don't need the values I think

Should be fixed now.

**Comment by angelozerr** [comment]
This PR shows Quarkus properties in flat mode in the Outline:

image

**Comment by fbricon** [comment]
Displaying the values seems redundant IMHO. The outline is supposed to give an overview of the doc, not contain the doc itself

**Comment by angelozerr** [comment]
The outline remove the comments and empty lines.

> Displaying the values seems redundant IMHO. The outline is supposed to give an overview of the doc, not contain the doc itself

What is your suggestion please?

Please note that I would like to display properties as a tree https://github.com/redhat-developer/quarkus-ls/issues/72

**Comment by angelozerr** [comment]
@fbricon I removed the =value. Here the result:

image

**Pull request merged**